src/ima-engine-decorator.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+import {BaseEngineDecorator, FakeEvent} from '@playkit-js/playkit-js';
+import {Ima} from './ima';
+
+/**
+ * Engine decorator for ima plugin.
+ * @class ImaEngineDecorator
+ * @param {IEngine} engine - The HTML5 engine.
+ * @param {Ima} plugin - The ima plugin.
+ */
+class ImaEngineDecorator extends BaseEngineDecorator {
+  _plugin: Ima;
+
+  constructor(engine: IEngine, plugin: Ima) {
+    super(engine);
+    this._plugin = plugin;
+  }
+
+  dispatchEvent(event: FakeEvent): boolean {
+    if (this._plugin.isAdsPlayingCustomPlayback()) {
+      return event.defaultPrevented;
+    } else {
+      return super.dispatchEvent(event);
+    }
+  }
+  /**
+   * Get paused state.
+   * @returns {boolean} - The paused value of the engine.
+   * @public
+   * @override
+   * @instance
+   * @memberof ImaEngineDecorator
+   */
+  get paused(): boolean {
+    if (!this._plugin.isAdsPlayingCustomPlayback()) {
```

Comment:
I would write it:
`return this._plugin.isAdsPlayingCustomPlayback() ? true : super.paused`;